dbt/include/firebolt/macros/materializations/models/incremental/incremental.sql

```diff
@@ -19,71 +19,80 @@
-     If `this` is None, returns None.
-     Note that this does *not* create a table in the DB.
-     We're doing this in case `this` is a view. Now we have a relation
-     with all the same field values, but as a table. #}
-  {% set source = this.incorporate(type='table') %}
-  {# If a table with the name `this.identifier` already exists, load_relation()
-     returns a BaseRelation with the dictionary values of that table, else
-     None. Note that, as with incorporate(), this does *not* create a table
-     in the DB. #}
-  {% set existing = load_relation(this) %}
-  {% set new_records = make_temp_relation(source) %}
-  {{ drop_relation_if_exists(new_records) }}
-  {# Out of an abundance of caution, setting type to view and dropping,
-     the setting it back to table. #}
-  {% set new_records = new_records.incorporate(type='view') %}
+  {%- set strategy = config.get('incremental_strategy') -%}
+  {%- if is_incremental() and strategy is none -%}
+    {{ log('Model %s is set to incremental, but no incremental strategy is set. '
+           'Defaulting to append' % this, True) }}
+    {%- set strategy = 'append' -%}
+  {%- endif -%}
+  {# In following lines:
+
+     `target` is just a dbt `BaseRelation` object, not a DB table.
+
+     We're only retrieving `existing` to check for existence; `load_relation()`
+     returns a `BaseRelation` with the dictionary values of any relations that exist
+     in dbt's cache that share the identifier of the passed relation. If none
+     exist, returns None. Note that this does *not* create a table in the DB.
+
+     `target` is a relation with all the same fields as `existing`, but guaranteed
+     to be an actual `BaseRelation` object. #}
+  {%- set target = this.incorporate(type='table') -%}
+  {%- set existing = load_relation(this) -%}
+  {%- set new_records = make_temp_relation(target) -%}
   {{ drop_relation_if_exists(new_records) }}
-  {% set new_records = new_records.incorporate(type='table') %}
-  {% set on_schema_change = incremental_validate_on_schema_change(
+  {%- set new_records = new_records.incorporate(type='table') -%}
+  {%- set on_schema_change = incremental_validate_on_schema_change(
                                 config.get('on_schema_change'),
-                                default='ignore') %}
+                                default='ignore') -%}
 
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
-
+  {%- set partition_by = config.get('partition_by') -%}
   {# First check whether we want to full refresh for existing view or config reasons. #}
-  {% set do_full_refresh = (should_full_refresh() or existing.is_view) %}
+  {%- set do_full_refresh = (should_full_refresh()
+                            or existing.is_view
+                            or (strategy == 'insert_overwrite' and not partition_by)) %}
+  {%- if strategy == 'insert_overwrite' and not partition_by -%}
+    {{ log('`partition_by` is not specified for model %s. This '
```

Comment:
I think you should raise an exception if the strategy is insert_overwrite and a partition isn't defined

Reply:
@octavianzarzu30 might disagree, but I would prefer to get an error instead of only realizing there was an issue on the second run

Reply:
Oh and I would make sure that it raises the exception at dbt compile time and not run time